pkg/controller/controller.go

```diff
@@ -56,7 +56,37 @@ import (
 	utilfeature "k8s.io/apiserver/pkg/util/feature"
 )
 
+type secretParamsType struct {
```

Comment:
maybe consider to change the name because it is a mapping from deprecated parameters to new ones instead of just secretParameter